crates/rune-cli/src/doc.rs

```diff
@@ -103,6 +123,7 @@ fn walk_items(io: &mut Io<'_>, doc: &DocFinder, queue: &mut VecDeque<ItemBuf>) -
 struct DocFinder {
     meta: BTreeMap<ItemBuf, MetaKind>,
     docs: HashMap<ItemBuf, Vec<String>>,
+    field_docs: HashMap<ItemBuf, HashMap<Box<str>, Vec<String>>>,
```

Comment:
Potential conflict when a variant's path and a struct's path happen to collide.

Reply:
Yeah, this should be disallowed on a language level, but currently it uh, does something weird (spins forever) so I just found a bug :):

```rust
enum Foo {
    /// Hello
    Variant,
}

mod Foo {
    //// Hello
    struct Variant;
}
```

In Rust, this would give and we should replicate it:

```
1 | enum Foo {
  | -------- previous definition of the type `Foo` here
...
6 | mod Foo {
  | ^^^^^^^ `Foo` redefined here
  |
  = note: `Foo` must be defined only once in the type namespace of this module
```

Unless you can think of another way this can conflict?

Reply:
This setup properly crashes with the below message when rolling back to bd33c6b796f676242007fdc1d6bf0fb4559fbc13. Trying to narrow it down further, at the moment.

```
error: query error
   ┌─ main:24:13
   │    
24 │ ╭ ╭             struct Foo {
25 │ │ │                 /// Struct field A.
26 │ │ │                 a,
27 │ │ │                 /// Struct field B.
28 │ │ │                 b,
29 │ │ │             }
   │ ╰─│─────────────^ `Enum::Foo` can refer to multiple things
   │   ╰─────────────' here as `Enum::Foo`
   · │ │
33 │                 Foo {
   │                 --- here as `Enum::Foo`
```

Reply:
And yeah, that was the way I thought it'd conflict. I wasn't able to reach the above crash and deem the conflict impossible at this stage due to the spinning forever thing.

Reply:
Spins after 003a32b0990ddd7bd25b5d69e02071674c25c8c9.

Reply:
I think this was due to a change a while back which allows for multiple items to be indexed (to improve diagnostics), but eventually only one should be allowed to be inserted unambiguously.

The issue arises when the meta cache is queried before exhausting indexed entries, causing the loop that you identified on Discord to never exit. This should be fixed by #417.